src/libraries/Directory.Build.targets

```diff
@@ -295,4 +296,24 @@
     <DefaultValidateFramework Remove="net6.0" />
   </ItemGroup>
 
+  <PropertyGroup Condition="Exists('$(PkgProjPath)')"> 
+    <ServicingVersion Condition="'$(ServicingVersion)' == ''">0</ServicingVersion>
+    <!-- Always update the package version.-->
+    <PackageVersion>$(MajorVersion).$(MinorVersion).$(ServicingVersion)</PackageVersion>
+  </PropertyGroup>
+  
+  <PropertyGroup Condition="Exists('$(PkgProjPath)') and '$(AssemblyVersion)' == '$(MajorVersion).$(MinorVersion).0.0'">       
+    <AssemblyInNetCoreApp Condition="$(NetCoreAppLibrary.Contains('$(AssemblyName);')) or $(NetCoreAppLibraryNoReference.Contains('$(AssemblyName);'))">true</AssemblyInNetCoreApp>
+    <AssemblyInAspNetCoreApp Condition="$(AspNetCoreAppLibrary.Contains('$(AssemblyName);'))">true</AssemblyInAspNetCoreApp>
+    <AssemblyInWindowsDesktop Condition="$(WindowsDesktopCoreAppLibrary.Contains('$(AssemblyName);'))">true</AssemblyInWindowsDesktop>
+    <AssemblyInRefPack  Condition="'$(AssemblyInNetCoreApp)' == 'true' or '$(AssemblyInAspNetCoreApp)' == 'true' or '$(AssemblyInWindowsDesktop)' == 'true'">true</AssemblyInRefPack>
+    <!-- Assembly version donot get updated in non-netfx ref pack assemblies. -->
+    <AssemblyVersion  Condition="'$(AssemblyInRefPack)' != 'true'">$(MajorVersion).$(MinorVersion).0.$(ServicingVersion)</AssemblyVersion>
+    <AssemblyVersion Condition="$(TargetFramework.StartsWith('net4')) and '$(AssemblyInRefPack)' == 'true'">$(MajorVersion).$(MinorVersion).0.$(ServicingVersion)</AssemblyVersion>
```

Comment:
Couldn't this be simplified by using a single prop definition and having a condition like:

`'$(AssemblyInRefPack)' != 'true' or $(TargetFramework.StartsWith('net4'))`